app/actions/index.js

```diff
@@ -53,27 +54,54 @@ export const logDataFeedResult = ({ dataFeedAddress }) => dispatch => {
 
 export const logMedianDataFeedResult = ({ dataFeedAddress }) => (dispatch, getState) => {
   let dataFeeds = getState().dataFeeds
-  _.map(dataFeeds, dataFeed => { dataFeed.currentResult = new Number(dataFeed.currentResult) })
-  dataFeeds = _.map(_.sortBy(dataFeeds, ['currentResult']), _.property('dataFeedAddress'))
+  _.map(dataFeeds, dataFeed => { dataFeed.lastUpdatedResult = new Number(dataFeed.lastUpdatedResult) })
+  dataFeeds = _.map(_.sortBy(dataFeeds, ['lastUpdatedResult']), _.property('dataFeedAddress'))
   return client.logMedianDataFeedResult(dataFeeds).then(
     () => {
       dispatch(fetchDataFeedLatestResult( { dataFeedAddress }))
     }
   )
 }
 
-export const dataFeedLatestResultLoaded = ({ currentResult, lastUpdated, dataFeedAddress }) => ({
+export const updateAllDataFeeds = ({ dataFeedAddrs, medianDataFeedAddress }) => (dispatch, getState) => {
+  return Promise.all(
+    _.map(dataFeedAddrs, dataFeedAddress => {
+      return dispatch(fetchDataFeedLatestResult({ dataFeedAddress }))
+    })
+  ).then(
+    () => {
+      dispatch(updateAll())
+    }
+  )
+}
+
+export const updateAll = () => (dispatch, getState) => {
```

Comment:
since this would probably not be used by itself, maybe put this logic into the `updateAllDataFeeds` action